docs/src/content/concepts-modes.md

```diff
@@ -138,6 +139,62 @@ most cases, the configuration will look like this:
 
 {{< figure src="/schematics/proxy-modes-transparent-3.png" >}}
 
+## WireGuard (transparent proxy)
+
+```shell
+mitmdump --mode wireguard
+```
+
+The WireGuard mode works in the same way as transparent mode, except that setup
+and routing client traffic to mitmproxy are different. In this mode, mitmproxy
+runs an internal WireGuard server, which devices can be connected to by using
+standard WireGuard clients applications, and just by importing the WireGuard
+configuration provided by mitmproxy.
+
+No additional routing configuration is required for this mode, since WireGuard
+operates by exchanging UDP packets between client and server (with encrypted IP
+packets as their payload) instead of routing IP packets directly. Additionally,
+the WireGuard server runs entirely in userspace, so no administrative privileges
+are necessary for setup or operation of mitmproxy in this mode.
+
+### Configuration
+
+By default, the WireGuard server will listen on port `51820/udp`, the default
+port for WireGuard servers, which should not require modifying firewall settings
+in most circumstances. However, changing firewall settings might be necessary
+when specifying a different port (i.e. when using something like
+`--mode wireguard@51821`).
+
+The encryption keys for WireGuard connections are stored in a file on disk, and
+it is possible to specify its path manually with `--mode wireguard:path`. If no
+path is specified, a fallback path within the default mitmproxy configuration
+directory is used. If no file exists at that path when the WireGuard server is
+instantiated, new keys will be generated and stored there automatically.
+
+### Limitations
+
+#### Transparently proxying mitmproxy host traffic
+
+With the current implementation, it is not possible to proxy all traffic of the
+host that mitmproxy itself is running on, since this would result in outgoing
+WireGuard packets being sent over the WireGuard tunnel themselves.
+
+However, it is possible to limit the IP addresses for which traffic is sent over
+the WireGuard tunnel to specific ranges. In this case, the `AllowedIPs` setting
+in the WireGuard client configuration needs to be changed from `0.0.0.0/0` (i.e.
+"route *all* IPv4 traffic through WireGuard tunnel") to the required ranges of
+IP addresses (this setting allows multiple, comma-separated values).
```

Comment:
That does not really help by itself, no? The problem is that you additionally need to redirect, otherwise it's caught again. I'd just skip this for now. The great plan is to have dedicated "current OS proxy" modes anyhow.

Reply:
You're right. I got confused because this is how it works for the echo server, but that of course doesn't help for transparent proxy. I'll remove that paragraph.